dtale/cli/loaders/excel_loader.py

```diff
@@ -0,0 +1,77 @@
+import pandas as pd
+import requests
+from six import PY3, BytesIO, StringIO
+
+from dtale.app import show
+from dtale.cli.clickutils import get_loader_options, loader_prop_keys
+
+"""
+  IMPORTANT!!! These global variables are required for building any customized CLI loader.
+  When build_loaders runs startup it will search for any modules containing the global variable LOADER_KEY.
+"""
+LOADER_KEY = "excel"
+LOADER_PROPS = [
+    dict(name="path", help="path to Excel (.xls,xlsx) file"),
+    dict(name="proxy", help="proxy URL if you're passing in a URL for --excel-path"),
+    dict(
+        name="parse_dates",
+        help="comma-separated string of column names which should be parsed as dates",
+    ),
+    dict(name="index_col", help="Column(s) to use as the row labels or the Datafame"),
+    dict(name="sheet", help="Name of the sheet to load from your Excel file"),
+]
+
+
+# IMPORTANT!!! This function is required if you would like to be able to use this loader from the back-end.
+def show_loader(**kwargs):
+    return show(data_loader=lambda: loader_func(**kwargs), **kwargs)
+
+
+def loader_func(**kwargs):
+    path = kwargs.pop("path")
+    if path.startswith("http://") or path.startswith("https://"):
+        engine = "xlrd" if path.endswith("xls") else "openpyxl"
+        sheet_name = kwargs.pop("sheet", None)
+        proxy = kwargs.pop("proxy", None)
+        req_kwargs = {}
+        if proxy is not None:
+            req_kwargs["proxies"] = dict(http=proxy, https=proxy)
+        resp = requests.get(path, **req_kwargs)
+        assert resp.status_code == 200
+        path = BytesIO(resp.content) if PY3 else StringIO(resp.content.decode("utf-8"))
+        dfs = pd.read_excel(
+            path, sheet_name=sheet_name,engine=engine,
+            **{k: v for k, v in kwargs.items() if k in loader_prop_keys(LOADER_PROPS)}
+        )
+        if not dfs:
+            raise Exception("Failed to load Excel file. Returned no data.")
+        if sheet_name:
+            return dfs[sheet_name]
+        # this is required because there is no support for loading multiple datasets at once from the CLI
+        # I can add this later...
+    return dfs[list(dfs.keys())[0]]
```

Comment:
```suggestion
    if path.startswith("http://") or path.startswith("https://"):
        proxy = kwargs.pop("proxy", None)
        req_kwargs = {}
        if proxy is not None:
            req_kwargs["proxies"] = dict(http=proxy, https=proxy)
        resp = requests.get(path, **req_kwargs)
        assert resp.status_code == 200
        path = BytesIO(resp.content) if PY3 else StringIO(resp.content.decode("utf-8"))
    engine = "xlrd" if path.endswith("xls") else "openpyxl"
    sheet_name = kwargs.pop("sheet", None)
    dfs = pd.read_excel(
        path, sheet_name=sheet_name, engine=engine,
        **{k: v for k, v in kwargs.items() if k in loader_prop_keys(LOADER_PROPS)}
    )
    if not dfs:
        raise Exception("Failed to load Excel file. Returned no data.")
    if sheet_name:
        return dfs[sheet_name]
    # this is required because there is no support for loading multiple datasets at once from the CLI
    # I can add this later...
    return dfs[list(dfs.keys())[0]]
```

Reply:
I think this is the last change and then this PR is good to go! The way this is written right now it should fail for non-URL paths.

I love your idea for load data from s3:\\ although I worry that will also require some sort of authentication to work (but I'll leave that you 😉 ).

I also agree that there is redundancy in the loaders, but I left that just so it's clear what is needed so others can add their own.  If you can find some refactoring that can be done I'll be happy to review 🙏